app/models/manageiq/providers/redhat/infra_manager/vm_import.rb

```diff
@@ -116,4 +158,22 @@ def select_host(vm)
   def password_auth(provider)
     provider.authentication_userid_passwords.first
   end
+
+  def get_first_vm_network_id(conn, dc_id)
+    conn.system_service.data_centers_service.data_center_service(dc_id)
+        .networks_service.list.select { |net| net.usages.include? 'vm' }.first.id
+  end
+
+  def get_vnic_profile_id_by_network_id(conn, network_id)
+    conn.system_service.networks_service.network_service(network_id).vnic_profiles_service.list.first.id
+  end
+
+  def set_all_vm_nics_to_profile_id(conn, vm_id, profile_id)
+    vm_nics_service = conn.system_service.vms_service.vm_service(vm_id).nics_service
+    vm_nics_service.list.map(&:id).each do |nic_id|
+      vm_nics_service.nic_service(nic_id).update(
+        OvirtSDK4::Nic.new(:vnic_profile => OvirtSDK4::VnicProfile.new(:id => profile_id))
```

Comment:
This can be shortened, if you want:

```ruby
...update(
  vnic_profile: {
    id: profile_id
  }
)
```

The SDK translates that hash into the same object that you are explicitly creating. Up to you, both approaches are perfectly OK.

I also find the use of `.map(&:id)` a bit harder to read. Wouldn't it be easier to do this?

```ruby
vm_nics_service.list.each do |nic|
  vm_nics_service.nic_service(nic.id).update(
    ...
  )
end